sdk/inc/azure/core/internal/az_http_internal.h

```diff
@@ -225,15 +225,10 @@ AZ_NODISCARD az_result az_http_request_set_query_parameter(
  * @param name Header name (e.g. `"Content-Type"`).
  * @param value Header value (e.g. `"application/x-www-form-urlencoded"`).
  *
- * @return
- *   - *`AZ_OK`* success.
- *   - *`AZ_ERROR_INSUFFICIENT_SPAN_SIZE`* there isn't enough space in the `p_request->buffer`
- * to add a header.
- *   - *`AZ_ERROR_ARG`*
- *     - `ref_request` is _NULL_.
- *     - `name` or `value` are invalid spans (see @ref _az_span_is_valid).
- *     - `name` or `value` are empty.
- *     - `name`'s or `value`'s buffer overlap resulting `url`'s buffer.
```

Comment:
So, we never return this `az_result`? More reason to update the docs :)

Reply:
What does actually happen if name/value are empty or overlapping?

Reply:
Yes, we never return that here anymore, we have them as precondition now. For overlapping regions, that could've been a problem in the future, and since then the function was rewritten. Maybe it is still a problem, but it is not ovious from the implementation at the very least to a point that without going an checking all the function calls, it seems that the code would handle the overlapping correctly. Not necessarily the truth, but that's how it looks like. And the rest are preconditions. Btw, we do have a `_az_PRECONDITION_NO_OVERLAP_SPANS()` if we wanted to check for this, but again - from the implementation I it is not apparent that overlapping would create a problem.